### PR TITLE
Clean up code in all-hands-demo branch

### PR DESCRIPTION
**Pull request by chromium-52**
# Pull Request

Trello Ticket Title: [Store And Checkout Cleanup](https://trello.com/c/lyFTr5Uw/54-store-and-checkout-cleanup)

---

## Why are these changes needed?

To clean up code in the all-hands-demo branch.

## What changed?

* Remove states in `QuantityUpdate`
* Combine `QuantityUpdate` and `LineItemUpdateQuantity` components
* Add `Title` component

## How are these changes tested?

Visual verification that all interactions behave as expected (e.g. increasing and decreasing items in the Store and Checkout pages).

Screenshots:

image

image

### CODE REVIEW

File: src/components/LineItem/LineItem.styles.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  paddingBottom: 2,
+  paddingLeft: 12,
+  paddingRight: 12,
+} as const;
+
+export const stylesButton = StyleSheet.create({
+  container: buttonContainerStyles,
+  button: buttonStyles,
+  text: textStyles,
+});
```
Comment (aryanshah701) [inline src/components/LineItem/LineItem.styles.ts:66]: i don't see button styles being used anywhere? we should remove it if it isn't being used

File: src/components/QuantityUpdate/QuantityUpdate.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  const onDecrease = () => {
+    if (shopState.checkout && lineItem) {
+      const newQuantity = lineItem.quantity - 1;
+      return newQuantity === 0
+        ? removeItemFromCheckout()
+        : updateCheckout(newQuantity);
+    } else {
+      return <></>;
+    }
+  };
```
Comment (aryanshah701) [inline src/components/QuantityUpdate/QuantityUpdate.tsx:81]: the same changes for onDecrease as suggested for onIncrease

File: src/components/Title/Title.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { styles } from './title.styles';
+
+type TitleProps = {
+  title: string;
+  style?: any;
+};
+
+export const Title: React.FC<TitleProps> = ({ title, style }) => {
+  return <Text style={[styles.title, style]}>{title}</Text>;
+};
```
Comment (aryanshah701) [inline src/components/Title/Title.tsx:12]: nice!

File: src/context/types.ts [hunk 4]
```diff
@@ -1,5 +1,13 @@
+export interface LineItem {
+  title: string;
+  totalPrice: number;
+  quantity: number;
+  priceUnit: number;
+}
```
Comment (aryanshah701) [inline src/context/types.ts:6]: let's move this to the LineItem component types file and rename it to LineItemType?

File: src/screens/Checkout/Checkout.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import { SetStateType } from '../../context/types';
+import { Title } from '../../components/Title';
+
+export const CheckoutScreen: React.FC<{
+  setState: SetStateType<ShopContextType>;
+}> = ({ setState }) => {
+  const { checkout } = useShopContext();
+  const checkoutUrl = (checkout as any)?.webUrl;
+  const totalTax = (checkout as any)?.totalTax;
+  const totalPrice = (checkout as any)?.totalPrice;
```
Comment (aryanshah701) [inline src/screens/Checkout/Checkout.tsx:15]: for now, let's make our own ShopifyCart type in the Checkout screen types file. You can simply make the type like this for now 
```
interface ShopifyCart {
   webUrl: string;
   totalTax: number;
   totalPrice: number;
   [x: string]: any; // this line allows any other key value pairs
}
```
then we can remove any code that does `(checkout as any)`. As the project grows we'll start typing the Cart interface with the appropriate fields, which we'll be able to do without having to make changes anywhere else.

You may run into a types issue with the all the service functions that create a checkout and update it. Simply cast the checkout to unknown and then our ShopifyCart type like so `setState({...state, (checkout as unknown as ShopifyCart)})`

File: src/screens/Store/Store.tsx [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
 
-type StoreScreenNavigationProp = BottomTabNavigationProp<
-  NavBarParamList,
-  'Store'
->;
-
-export const StoreScreen: React.FC = () => {
-  const navigation = useNavigation<StoreScreenNavigationProp>();
+export const StoreScreen: React.FC<{
+  setState: SetStateType<ShopContextType>;
```
Comment (aryanshah701) [inline src/screens/Store/Store.tsx:10]: lets not inline type the props and create a StoreScreenProps type with the setState prop

File: src/screens/Checkout/Checkout.tsx (hunk 5 above)
Comment (gerrifox28) [reply src/screens/Checkout/Checkout.tsx:15]: the service functions didn't seem to give me any errors but let me know if I should still update them

**Review by aryanshah701 — APPROVED** [review]

**Pull request merged**